Tests fail on s390x architecture

#### Describe the bug
Some tests fail on the s390x architecture. Since s390x is a big-endian arch, it parses the floats/integers in an opposite way as the little-endian arch does.

#### Steps to reproduce
Simply run the tests on a big-endian architecture machine.
List of tests failing: 

```
FAILED mne/channels/tests/test_layout.py::test_find_layout
FAILED mne/io/edf/tests/test_edf.py::test_edf_data_broken
FAILED mne/io/edf/tests/test_edf.py::test_parse_annotation
FAILED mne/io/edf/tests/test_edf.py::test_find_events_backward_compatibility
FAILED mne/io/edf/tests/test_edf.py::test_read_annot
FAILED mne/io/egi/tests/test_egi.py::test_io_egi
FAILED mne/io/kit/tests/test_coreg.py::test_io_mrk
FAILED mne/io/kit/tests/test_kit.py::test_epochs
FAILED mne/io/kit/tests/test_kit.py::test_raw_events
FAILED mne/io/kit/tests/test_kit.py::test_ch_loc
FAILED mne/io/kit/tests/test_kit.py::test_hsp_elp
FAILED mne/io/kit/tests/test_kit.py::test_decimate
FAILED mne/io/tests/test_pick.py::test_pick_refs
FAILED mne/io/tests/test_read_raw.py::test_read_raw_supported[fname3]
FAILED mne/preprocessing/tests/test_maxwell.py::test_other_systems
```

#### Expected results
Tests should pass


#### Additional information
Inside the tests, we could do a quick check to find if the machine is of little-endian arch or big-endian, using the sys library.

```python
>>> import sys
>>> sys.byteorder
'little'
```

And then, simply change the way how we parse an integer when reading from a testfile. For example, one of the errors occur because of the below line.

```python
size=np.fromfile(fid, np.int32, 1)[0]
```

so, we can replace this with:

```python
import sys

little_end = False
if sys.byteorder == "little":
    little_end = True

...

if little_end:
    size=np.fromfile(fid, np.dtype('i4'), 1)[0]
    # or explicitly mention little-endian byte order
    # i4 means the int32 dtype, and appending a "<" or a ">" means a little-endian or a big-endian dtype
    size=np.fromfile(fid, np.dtype('<i4'), 1)[0]
else:
    size=np.fromfile(fid, np.dtype('>i4'), 1)[0]
...
```

#### Actual results
The whole build log file can be found (big-endian) [here](https://kojipkgs.fedoraproject.org//work/tasks/629/56760629/build.log)

[Here](https://kojipkgs.fedoraproject.org//work/tasks/5769/56755769/build.log) is a build from a little-endian machine.

I can probably get access to an s390x arch machine if needed for debugging or testing purposes.

Some links about the big-endian architecture and numpy:

1. https://github.com/numpy/numpy/issues/13638
2. https://stackoverflow.com/questions/9056373/what-is-the-difference-between-big-and-little-endian-floats#9056399
3. https://numpy.org/doc/stable/reference/arrays.dtypes.html

Info about failing tests, copy-pasted below:

```
=================================== FAILURES ===================================
_______________________________ test_find_layout _______________________________
mne/channels/tests/test_layout.py:255: in test_find_layout
    raw_kit = read_raw_kit(fname_kit_157)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
    Read a total of 3 projection items:
        PCA-v1 (1 x 102)  idle
        PCA-v2 (1 x 102)  idle
        PCA-v3 (1 x 102)  idle
    Read 5 compensation matrices
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/channels/tests/../../io/kit/tests/data/test.sqd...
Creating Raw.info structure...
_____________________________ test_edf_data_broken _____________________________
mne/io/edf/tests/test_edf.py:130: in test_edf_data_broken
    raw = _test_raw_reader(read_raw_edf, input_fname=edf_path,
mne/io/tests/test_raw.py:189: in _test_raw_reader
    assert_array_less(data, maxval)
E   AssertionError: 
E   Arrays are not less-ordered
E   
E   Mismatched elements: 12283 / 417792 (2.94%)
E   Max absolute difference: 0.05090099
E   Max relative difference: 5.09009855
E    x: array([[0.001702, 0.000423, 0.000769, ..., 0.002524, 0.001299, 0.000917],
E          [0.002138, 0.000244, 0.000334, ..., 0.001169, 0.000199, 0.000183],
E          [0.004641, 0.000301, 0.000377, ..., 0.002903, 0.001678, 0.001296],...
E    y: array(0.01)
____________________________ test_parse_annotation _____________________________
mne/io/edf/tests/test_edf.py:197: in test_parse_annotation
    assert_equal(np.column_stack((onset, duration, description)),
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (3, 3), (6, 3) mismatch)
E    x: array([['1.0', '4402.0', '0onhtni'],
E          ['1.0', '4402.0', 'g\x00\x00\x00\x001+08.0\x152525.A'],
E          ['1.0', '4402.0', 'npae']], dtype='>U32')
E    y: array([['180.0', '0.0', 'Lights off'],
E          ['180.0', '0.0', 'Close door'],
E          ['180.0', '0.0', 'Lights off'],...
___________________ test_find_events_backward_compatibility ____________________
mne/io/edf/tests/test_edf.py:212: in test_find_events_backward_compatibility
    event_id.pop('start')
E   KeyError: 'start'
----------------------------- Captured stdout call -----------------------------
Extracting EDF parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/edf/tests/data/test.edf...
EDF file detected
Setting channel info structure...
Creating raw.info structure...
Reading 0 ... 3071  =      0.000 ...     5.998 secs...
_______________________________ test_read_annot ________________________________
mne/io/edf/tests/test_edf.py:292: in test_read_annot
    _assert_annotations_equal(annotation, EXPECTED_ANNOTATIONS)
mne/io/edf/tests/test_edf.py:251: in _assert_annotations_equal
    assert_array_equal(a.onset, b.onset)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (3,), (6,) mismatch)
E    x: array([1., 1., 1.])
E    y: array([   3.14,  180.  ,  180.  ,  180.  ,  180.  , 1800.2 ])
_________________________________ test_io_egi __________________________________
mne/io/egi/tests/test_egi.py:143: in test_io_egi
    raw = read_raw_egi(egi_fname, include=None)
<decorator-gen-182>:24: in read_raw_egi
    ???
mne/io/egi/egi.py:153: in read_raw_egi
    return RawEGI(input_fname, eog, misc, include, exclude, preload,
<decorator-gen-183>:24: in __init__
    ???
mne/io/egi/egi.py:170: in __init__
    egi_info = _read_header(fid)
mne/io/egi/egi.py:52: in _read_header
    raise RuntimeError('Floating point precision is undefined.')
E   RuntimeError: Floating point precision is undefined.
----------------------------- Captured stdout call -----------------------------
Reading EGI header from /builddir/build/BUILD/mne-python-0.21.2/mne/io/egi/tests/data/test_egi.raw...
_________________________________ test_io_mrk __________________________________
mne/io/kit/tests/test_coreg.py:26: in test_io_mrk
    pts = read_mrk(mrk_fname)
mne/io/kit/coreg.py:40: in read_mrk
    mrk_count = unpack('i', fid.read(KIT.INT))[0]
E   struct.error: unpack requires a buffer of 4 bytes
_________________________________ test_epochs __________________________________
mne/io/kit/tests/test_kit.py:205: in test_epochs
    raw = read_raw_kit(sqd_path, stim=None)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
_______________________________ test_raw_events ________________________________
mne/io/kit/tests/test_kit.py:222: in test_raw_events
    raw = read_raw_kit(sqd_path)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
_________________________________ test_ch_loc __________________________________
mne/io/kit/tests/test_kit.py:246: in test_ch_loc
    raw_py = read_raw_kit(sqd_path, mrk_path, elp_txt_path, hsp_txt_path,
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
_________________________________ test_hsp_elp _________________________________
mne/io/kit/tests/test_kit.py:275: in test_hsp_elp
    raw_txt = read_raw_kit(sqd_path, mrk_path, elp_txt_path, hsp_txt_path)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
________________________________ test_decimate _________________________________
mne/io/kit/tests/test_kit.py:308: in test_decimate
    raw = read_raw_kit(sqd_path, mrk_path, elp_txt_path, sphere_hsp_path)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
________________________________ test_pick_refs ________________________________
mne/io/tests/test_pick.py:99: in test_pick_refs
    raw_kit = read_raw_kit(sqd_path, mrk_path, elp_path, hsp_path)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/tests/../kit/tests/data/test.sqd...
Creating Raw.info structure...
_______________________ test_read_raw_supported[fname3] ________________________
mne/io/tests/test_read_raw.py:35: in test_read_raw_supported
    read_raw(fname)
mne/io/_read_raw.py:82: in read_raw
    return readers[ext](fname, preload=preload, verbose=verbose, **kwargs)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/io/kit/tests/data/test.sqd...
Creating Raw.info structure...
______________________________ test_other_systems ______________________________
mne/preprocessing/tests/test_maxwell.py:235: in test_other_systems
    raw_kit = read_raw_kit(sqd_path, mrk_path, elp_path, hsp_path)
mne/io/kit/kit.py:903: in read_raw_kit
    return RawKIT(input_fname=input_fname, mrk=mrk, elp=elp, hsp=hsp,
<decorator-gen-211>:24: in __init__
    ???
mne/io/kit/kit.py:134: in __init__
    info, kit_info = get_kit_info(
<decorator-gen-213>:24: in get_kit_info
    ???
mne/io/kit/kit.py:495: in get_kit_info
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:495: in <genexpr>
    dirs.extend(_read_dir(fid) for _ in range(dirs[0]['count'] - 1))
mne/io/kit/kit.py:459: in _read_dir
    size=np.fromfile(fid, np.int32, 1)[0],
E   IndexError: index 0 is out of bounds for axis 0 with size 0
----------------------------- Captured stdout call -----------------------------
Extracting SQD Parameters from /builddir/build/BUILD/mne-python-0.21.2/mne/preprocessing/tests/../../io/kit/tests/data/test.sqd...
Creating Raw.info structure...
=============================== warnings summary ===============================
mne/utils/docs.py:1776
  /builddir/build/BUILD/mne-python-0.21.2/mne/utils/docs.py:1776: DeprecationWarning: `unindent_dict` is deprecated!
  scipy.misc.unindent_dict is deprecated in SciPy 1.3.0
    docdict = unindent_dict(docdict)
mne/utils/docs.py:1777
  /builddir/build/BUILD/mne-python-0.21.2/mne/utils/docs.py:1777: DeprecationWarning: `filldoc` is deprecated!
  scipy.misc.filldoc is deprecated in SciPy 1.3.0
    fill_doc = filldoc(docdict, unindent_params=False)
mne/datasets/utils.py:62
  /builddir/build/BUILD/mne-python-0.21.2/mne/datasets/utils.py:62: DeprecationWarning: `docformat` is deprecated!
  scipy.misc.docformat is deprecated in Scipy 1.3.0
    _data_path_doc = docformat(_data_path_doc, docdict)
mne/datasets/utils.py:63
  /builddir/build/BUILD/mne-python-0.21.2/mne/datasets/utils.py:63: DeprecationWarning: `docformat` is deprecated!
  scipy.misc.docformat is deprecated in Scipy 1.3.0
    _data_path_doc_accept = docformat(_data_path_doc_accept, docdict)
mne/conftest.py:449
  /builddir/build/BUILD/mne-python-0.21.2/mne/conftest.py:449: PytestUnknownMarkWarning: Unknown pytest.mark.slowtest - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
    @pytest.mark.slowtest
-- Docs: https://docs.pytest.org/en/stable/warnings.html
- generated xml file: /builddir/build/BUILD/mne-python-0.21.2/junit-results.xml -
```

--- a/doc/changes/latest.inc
+++ b/doc/changes/latest.inc
@@ -95,6 +95,8 @@ Bugs
 
 - Fix bug with reading split files that have dashes in the filename (:gh:`8339` **by new contributor** |Eduard Ort|_)
 
+- Fix bug with reading EDF and KIT files on big endian architectures such as s390x (:gh:`8618` by `Eric Larson`_)
+
 - Fix bug with parsing EDF dates and date integers (:gh:`8558` **by new contributor** |Austin Hurst|_ and `Eric Larson`_)
 
 - Fix bug with `~mne.viz.plot_epochs_image` when ``order`` is supplied and multiple conditions are plotted (:gh:`8377` by `Daniel McCloy`_ )

--- a/doc/changes/latest.inc
+++ b/doc/changes/latest.inc
@@ -95,9 +95,9 @@ Bugs
 
 - Fix bug with reading split files that have dashes in the filename (:gh:`8339` **by new contributor** |Eduard Ort|_)
 
-- Fix bug with reading EDF and KIT files on big endian architectures such as s390x (:gh:`8618` by `Eric Larson`_)
-
 - Fix bug with parsing EDF dates and date integers (:gh:`8558` **by new contributor** |Austin Hurst|_ and `Eric Larson`_)
+
+- Fix bug with reading EDF and KIT files on big endian architectures such as s390x (:gh:`8618` by `Eric Larson`_)
 
 - Fix bug with `~mne.viz.plot_epochs_image` when ``order`` is supplied and multiple conditions are plotted (:gh:`8377` by `Daniel McCloy`_ )
 

--- a/mne/io/edf/edf.py
+++ b/mne/io/edf/edf.py
@@ -218,7 +218,7 @@ def _read_ch(fid, subtype, samp, dtype_byte, dtype=None):
     # BDF
     if subtype == 'bdf':
         ch_data = np.fromfile(fid, dtype=dtype, count=samp * dtype_byte)
-        ch_data = ch_data.reshape(-1, 3).astype(np.int32)
+        ch_data = ch_data.reshape(-1, 3).astype(INT32)
         ch_data = ((ch_data[:, 0]) +
                    (ch_data[:, 1] << 8) +
                    (ch_data[:, 2] << 16))
@@ -671,17 +671,27 @@ def _read_edf_header(fname, exclude):
 
         if subtype == 'bdf':
             edf_info['dtype_byte'] = 3  # 24-bit (3 byte) integers
-            edf_info['dtype_np'] = np.uint8
+            edf_info['dtype_np'] = UINT8
         else:
             edf_info['dtype_byte'] = 2  # 16-bit (2 byte) integers
-            edf_info['dtype_np'] = np.int16
+            edf_info['dtype_np'] = INT16
 
     return edf_info, orig_units
 
 
-GDFTYPE_NP = (None, np.int8, np.uint8, np.int16, np.uint16, np.int32,
-              np.uint32, np.int64, np.uint64, None, None, None, None,
-              None, None, None, np.float32, np.float64)
+INT8 = '<i1'
+UINT8 = '<u1'
+INT16 = '<i2'
+UINT16 = '<u2'
+INT32 = '<i4'
+UINT32 = '<u4'
+INT64 = '<i8'
+UINT64 = '<u8'
+FLOAT32 = '<f4'
+FLOAT64 = '<f8'
+GDFTYPE_NP = (None, INT8, UINT8, INT16, UINT16, INT32, UINT32,
+              INT64, UINT64, None, None, None, None,
+              None, None, None, FLOAT32, FLOAT64)
 GDFTYPE_BYTE = tuple(np.dtype(x).itemsize if x is not None else 0
                      for x in GDFTYPE_NP)
 
@@ -737,20 +747,20 @@ def _read_gdf_header(fname, exclude):
             except Exception:
                 pass
 
-            header_nbytes = np.fromfile(fid, np.int64, 1)[0]
-            meas_id['equipment'] = np.fromfile(fid, np.uint8, 8)[0]
-            meas_id['hospital'] = np.fromfile(fid, np.uint8, 8)[0]
-            meas_id['technician'] = np.fromfile(fid, np.uint8, 8)[0]
+            header_nbytes = np.fromfile(fid, INT64, 1)[0]
+            meas_id['equipment'] = np.fromfile(fid, UINT8, 8)[0]
+            meas_id['hospital'] = np.fromfile(fid, UINT8, 8)[0]
+            meas_id['technician'] = np.fromfile(fid, UINT8, 8)[0]
             fid.seek(20, 1)    # 20bytes reserved
 
-            n_records = np.fromfile(fid, np.int64, 1)[0]
+            n_records = np.fromfile(fid, INT64, 1)[0]
             # record length in seconds
-            record_length = np.fromfile(fid, np.uint32, 2)
+            record_length = np.fromfile(fid, UINT32, 2)
             if record_length[0] == 0:
                 record_length[0] = 1.
                 warn('Header information is incorrect for record length. '
                      'Default record length set to 1.')
-            nchan = np.fromfile(fid, np.uint32, 1)[0]
+            nchan = np.fromfile(fid, UINT32, 1)[0]
             channels = list(range(nchan))
             ch_names = [fid.read(16).decode('latin-1').strip(' \x00')
                         for ch in channels]
@@ -770,19 +780,19 @@ def _read_gdf_header(fname, exclude):
             edf_info['units'] = np.array(edf_info['units'], float)
 
             ch_names = [ch_names[idx] for idx in sel]
-            physical_min = np.fromfile(fid, np.float64, len(channels))
-            physical_max = np.fromfile(fid, np.float64, len(channels))
-            digital_min = np.fromfile(fid, np.int64, len(channels))
-            digital_max = np.fromfile(fid, np.int64, len(channels))
+            physical_min = np.fromfile(fid, FLOAT64, len(channels))
+            physical_max = np.fromfile(fid, FLOAT64, len(channels))
+            digital_min = np.fromfile(fid, INT64, len(channels))
+            digital_max = np.fromfile(fid, INT64, len(channels))
             prefiltering = [fid.read(80).decode().strip(' \x00')
                             for ch in channels][:-1]
             highpass, lowpass = _parse_prefilter_string(prefiltering)
 
             # n samples per record
-            n_samps = np.fromfile(fid, np.int32, len(channels))
+            n_samps = np.fromfile(fid, INT32, len(channels))
 
             # channel data type
-            dtype = np.fromfile(fid, np.int32, len(channels))
+            dtype = np.fromfile(fid, INT32, len(channels))
 
             # total number of bytes for data
             bytes_tot = np.sum([GDFTYPE_BYTE[t] * n_samps[i]
@@ -809,22 +819,22 @@ def _read_gdf_header(fname, exclude):
             etp = header_nbytes + n_records * edf_info['bytes_tot']
             # skip data to go to event table
             fid.seek(etp)
-            etmode = np.fromfile(fid, np.uint8, 1)[0]
+            etmode = np.fromfile(fid, UINT8, 1)[0]
             if etmode in (1, 3):
-                sr = np.fromfile(fid, np.uint8, 3)
+                sr = np.fromfile(fid, UINT8, 3)
                 event_sr = sr[0]
                 for i in range(1, len(sr)):
                     event_sr = event_sr + sr[i] * 2 ** (i * 8)
-                n_events = np.fromfile(fid, np.uint32, 1)[0]
-                pos = np.fromfile(fid, np.uint32, n_events) - 1  # 1-based inds
-                typ = np.fromfile(fid, np.uint16, n_events)
+                n_events = np.fromfile(fid, UINT32, 1)[0]
+                pos = np.fromfile(fid, UINT32, n_events) - 1  # 1-based inds
+                typ = np.fromfile(fid, UINT16, n_events)
 
                 if etmode == 3:
-                    chn = np.fromfile(fid, np.uint16, n_events)
-                    dur = np.fromfile(fid, np.uint32, n_events)
+                    chn = np.fromfile(fid, UINT16, n_events)
+                    dur = np.fromfile(fid, UINT32, n_events)
                 else:
                     chn = np.zeros(n_events, dtype=np.int32)
-                    dur = np.ones(n_events, dtype=np.uint32)
+                    dur = np.ones(n_events, dtype=UINT32)
                 np.maximum(dur, 1, out=dur)
                 events = [n_events, pos, typ, chn, dur]
 
@@ -846,20 +856,20 @@ def _read_gdf_header(fname, exclude):
             fid.seek(10, 1)  # 10bytes reserved
 
             # Smoking / Alcohol abuse / drug abuse / medication
-            sadm = np.fromfile(fid, np.uint8, 1)[0]
+            sadm = np.fromfile(fid, UINT8, 1)[0]
             patient['smoking'] = scale[sadm % 4]
             patient['alcohol_abuse'] = scale[(sadm >> 2) % 4]
             patient['drug_abuse'] = scale[(sadm >> 4) % 4]
             patient['medication'] = scale[(sadm >> 6) % 4]
-            patient['weight'] = np.fromfile(fid, np.uint8, 1)[0]
+            patient['weight'] = np.fromfile(fid, UINT8, 1)[0]
             if patient['weight'] == 0 or patient['weight'] == 255:
                 patient['weight'] = None
-            patient['height'] = np.fromfile(fid, np.uint8, 1)[0]
+            patient['height'] = np.fromfile(fid, UINT8, 1)[0]
             if patient['height'] == 0 or patient['height'] == 255:
                 patient['height'] = None
 
             # Gender / Handedness / Visual Impairment
-            ghi = np.fromfile(fid, np.uint8, 1)[0]
+            ghi = np.fromfile(fid, UINT8, 1)[0]
             patient['sex'] = gender[ghi % 4]
             patient['handedness'] = handedness[(ghi >> 2) % 4]
             patient['visual'] = scale[(ghi >> 4) % 4]
@@ -867,7 +877,7 @@ def _read_gdf_header(fname, exclude):
             # Recording identification
             meas_id = {}
             meas_id['recording_id'] = fid.read(64).decode().strip(' \x00')
-            vhsv = np.fromfile(fid, np.uint8, 4)
+            vhsv = np.fromfile(fid, UINT8, 4)
             loc = {}
             if vhsv[3] == 0:
                 loc['vertpre'] = 10 * int(vhsv[0] >> 4) + int(vhsv[0] % 16)
@@ -879,20 +889,20 @@ def _read_gdf_header(fname, exclude):
                 loc['size'] = 29
             loc['version'] = 0
             loc['latitude'] = \
-                float(np.fromfile(fid, np.uint32, 1)[0]) / 3600000
+                float(np.fromfile(fid, UINT32, 1)[0]) / 3600000
             loc['longitude'] = \
-                float(np.fromfile(fid, np.uint32, 1)[0]) / 3600000
-            loc['altitude'] = float(np.fromfile(fid, np.int32, 1)[0]) / 100
+                float(np.fromfile(fid, UINT32, 1)[0]) / 3600000
+            loc['altitude'] = float(np.fromfile(fid, INT32, 1)[0]) / 100
             meas_id['loc'] = loc
 
-            meas_date = np.fromfile(fid, np.uint64, 1)[0]
+            meas_date = np.fromfile(fid, UINT64, 1)[0]
             if meas_date != 0:
                 meas_date = (datetime(1, 1, 1, tzinfo=timezone.utc) +
                              timedelta(meas_date * pow(2, -32) - 367))
             else:
                 meas_date = None
 
-            birthday = np.fromfile(fid, np.uint64, 1).tolist()[0]
+            birthday = np.fromfile(fid, UINT64, 1).tolist()[0]
             if birthday == 0:
                 birthday = datetime(1, 1, 1, tzinfo=timezone.utc)
             else:
@@ -909,28 +919,28 @@ def _read_gdf_header(fname, exclude):
             else:
                 patient['age'] = None
 
-            header_nbytes = np.fromfile(fid, np.uint16, 1)[0] * 256
+            header_nbytes = np.fromfile(fid, UINT16, 1)[0] * 256
 
             fid.seek(6, 1)  # 6 bytes reserved
-            meas_id['equipment'] = np.fromfile(fid, np.uint8, 8)
-            meas_id['ip'] = np.fromfile(fid, np.uint8, 6)
-            patient['headsize'] = np.fromfile(fid, np.uint16, 3)
+            meas_id['equipment'] = np.fromfile(fid, UINT8, 8)
+            meas_id['ip'] = np.fromfile(fid, UINT8, 6)
+            patient['headsize'] = np.fromfile(fid, UINT16, 3)
             patient['headsize'] = np.asarray(patient['headsize'], np.float32)
             patient['headsize'] = np.ma.masked_array(
                 patient['headsize'],
                 np.equal(patient['headsize'], 0), None).filled()
-            ref = np.fromfile(fid, np.float32, 3)
-            gnd = np.fromfile(fid, np.float32, 3)
-            n_records = np.fromfile(fid, np.int64, 1)[0]
+            ref = np.fromfile(fid, FLOAT32, 3)
+            gnd = np.fromfile(fid, FLOAT32, 3)
+            n_records = np.fromfile(fid, INT64, 1)[0]
 
             # record length in seconds
-            record_length = np.fromfile(fid, np.uint32, 2)
+            record_length = np.fromfile(fid, UINT32, 2)
             if record_length[0] == 0:
                 record_length[0] = 1.
                 warn('Header information is incorrect for record length. '
                      'Default record length set to 1.')
 
-            nchan = np.fromfile(fid, np.uint16, 1)[0]
+            nchan = np.fromfile(fid, UINT16, 1)[0]
             fid.seek(2, 1)  # 2bytes reserved
 
             # Channels (variable header)
@@ -949,7 +959,7 @@ def _read_gdf_header(fname, exclude):
             - Decimal factors codes:
             https://sourceforge.net/p/biosig/svn/HEAD/tree/trunk/biosig/doc/DecimalFactors.txt
             """  # noqa
-            units = np.fromfile(fid, np.uint16, len(channels)).tolist()
+            units = np.fromfile(fid, UINT16, len(channels)).tolist()
             unitcodes = np.array(units[:])
             edf_info['units'] = list()
             for i, unit in enumerate(units):
@@ -971,34 +981,34 @@ def _read_gdf_header(fname, exclude):
             edf_info['units'] = np.array(edf_info['units'], float)
 
             ch_names = [ch_names[idx] for idx in sel]
-            physical_min = np.fromfile(fid, np.float64, len(channels))
-            physical_max = np.fromfile(fid, np.float64, len(channels))
-            digital_min = np.fromfile(fid, np.float64, len(channels))
-            digital_max = np.fromfile(fid, np.float64, len(channels))
+            physical_min = np.fromfile(fid, FLOAT64, len(channels))
+            physical_max = np.fromfile(fid, FLOAT64, len(channels))
+            digital_min = np.fromfile(fid, FLOAT64, len(channels))
+            digital_max = np.fromfile(fid, FLOAT64, len(channels))
 
             fid.seek(68 * len(channels), 1)  # obsolete
-            lowpass = np.fromfile(fid, np.float32, len(channels))
-            highpass = np.fromfile(fid, np.float32, len(channels))
-            notch = np.fromfile(fid, np.float32, len(channels))
+            lowpass = np.fromfile(fid, FLOAT32, len(channels))
+            highpass = np.fromfile(fid, FLOAT32, len(channels))
+            notch = np.fromfile(fid, FLOAT32, len(channels))
 
             # number of samples per record
-            n_samps = np.fromfile(fid, np.int32, len(channels))
+            n_samps = np.fromfile(fid, INT32, len(channels))
 
             # data type
-            dtype = np.fromfile(fid, np.int32, len(channels))
+            dtype = np.fromfile(fid, INT32, len(channels))
 
             channel = {}
-            channel['xyz'] = [np.fromfile(fid, np.float32, 3)[0]
+            channel['xyz'] = [np.fromfile(fid, FLOAT32, 3)[0]
                               for ch in channels]
 
             if edf_info['number'] < 2.19:
-                impedance = np.fromfile(fid, np.uint8,
+                impedance = np.fromfile(fid, UINT8,
                                         len(channels)).astype(float)
                 impedance[impedance == 255] = np.nan
                 channel['impedance'] = pow(2, impedance / 8)
                 fid.seek(19 * len(channels), 1)  # reserved
             else:
-                tmp = np.fromfile(fid, np.float32, 5 * len(channels))
+                tmp = np.fromfile(fid, FLOAT32, 5 * len(channels))
                 tmp = tmp[::5]
                 fZ = tmp[:]
                 impedance = tmp[:]
@@ -1036,27 +1046,27 @@ def _read_gdf_header(fname, exclude):
             fid.seek(etp)  # skip data to go to event table
             etmode = fid.read(1).decode()
             if etmode != '':
-                etmode = np.fromstring(etmode, np.uint8).tolist()[0]
+                etmode = np.fromstring(etmode, UINT8).tolist()[0]
 
                 if edf_info['number'] < 1.94:
-                    sr = np.fromfile(fid, np.uint8, 3)
+                    sr = np.fromfile(fid, UINT8, 3)
                     event_sr = sr[0]
                     for i in range(1, len(sr)):
                         event_sr = event_sr + sr[i] * 2**(i * 8)
-                    n_events = np.fromfile(fid, np.uint32, 1)[0]
+                    n_events = np.fromfile(fid, UINT32, 1)[0]
                 else:
-                    ne = np.fromfile(fid, np.uint8, 3)
+                    ne = np.fromfile(fid, UINT8, 3)
                     n_events = ne[0]
                     for i in range(1, len(ne)):
                         n_events = n_events + ne[i] * 2**(i * 8)
-                    event_sr = np.fromfile(fid, np.float32, 1)[0]
+                    event_sr = np.fromfile(fid, FLOAT32, 1)[0]
 
-                pos = np.fromfile(fid, np.uint32, n_events) - 1  # 1-based inds
-                typ = np.fromfile(fid, np.uint16, n_events)
+                pos = np.fromfile(fid, UINT32, n_events) - 1  # 1-based inds
+                typ = np.fromfile(fid, UINT16, n_events)
 
                 if etmode == 3:
-                    chn = np.fromfile(fid, np.uint16, n_events)
-                    dur = np.fromfile(fid, np.uint32, n_events)
+                    chn = np.fromfile(fid, UINT16, n_events)
+                    dur = np.fromfile(fid, UINT32, n_events)
                 else:
                     chn = np.zeros(n_events, dtype=np.uint32)
                     dur = np.ones(n_events, dtype=np.uint32)
@@ -1388,8 +1398,8 @@ def _read_annotations_edf(annotations):
         annotations = np.atleast_2d(annotations)
         for chan in annotations:
             this_chan = chan.ravel()
-            if this_chan.dtype == np.int32:  # BDF
-                this_chan.dtype = np.uint8
+            if this_chan.dtype == INT32:  # BDF
+                this_chan = this_chan.view(dtype=UINT8)
                 this_chan = this_chan.reshape(-1, 4)
                 # Why only keep the first 3 bytes as BDF values
                 # are stored with 24 bits (not 32)
@@ -1397,7 +1407,7 @@ def _read_annotations_edf(annotations):
                 # As ravel() returns a 1D array we can add all values at once
                 tals.extend(this_chan)
             else:
-                this_chan = chan.astype(int)
+                this_chan = chan.astype(np.int64)
                 # Exploit np vectorized processing
                 tals.extend(np.uint8([this_chan % 256, this_chan // 256])
                             .flatten('F'))

--- a/mne/io/edf/tests/test_edf.py
+++ b/mne/io/edf/tests/test_edf.py
@@ -203,8 +203,8 @@ def test_parse_annotation(tmpdir):
                              dtype=np.int64)
 
     with open(str(annot_file), 'rb') as fid:
-        # ch_data = np.fromfile(fid, dtype=np.int16, count=len(annot))
-        tal_channel_B = _read_ch(fid, subtype='EDF', dtype=np.int16,
+        # ch_data = np.fromfile(fid, dtype='<i2', count=len(annot))
+        tal_channel_B = _read_ch(fid, subtype='EDF', dtype='<i2',
                                  samp=(len(annot) - 1) // 2,
                                  dtype_byte='This_parameter_is_not_used')
 
@@ -301,7 +301,7 @@ def test_read_annot(tmpdir):
 
     # Now test when reading from buffer of data
     with open(str(annot_file), 'rb') as fid:
-        ch_data = np.fromfile(fid, dtype=np.int16, count=len(annot))
+        ch_data = np.fromfile(fid, dtype='<i2', count=len(annot))
     onset, duration, desc = _read_annotations_edf([ch_data])
     annotation = Annotations(onset=onset, duration=duration, description=desc,
                              orig_time=None)

--- a/mne/io/egi/egi.py
+++ b/mne/io/egi/egi.py
@@ -19,7 +19,7 @@ from ...utils import verbose, logger, warn
 
 def _read_header(fid):
     """Read EGI binary header."""
-    version = np.fromfile(fid, np.int32, 1)[0]
+    version = np.fromfile(fid, '<i4', 1)[0]
 
     if version > 6 & ~np.bitwise_and(version, 6):
         version = version.byteswap().astype(np.uint32)

--- a/mne/io/kit/coreg.py
+++ b/mne/io/kit/coreg.py
@@ -7,12 +7,14 @@
 from os import SEEK_CUR, path as op
 import pickle
 import re
-from struct import unpack
 
 import numpy as np
 
 from .constants import KIT
 from .._digitization import _read_dig_points
+
+
+INT32 = '<i4'
 
 
 def read_mrk(fname):
@@ -33,11 +35,11 @@ def read_mrk(fname):
     if ext in ('.sqd', '.mrk'):
         with open(fname, 'rb', buffering=0) as fid:
             fid.seek(192)
-            mrk_offset = unpack('i', fid.read(KIT.INT))[0]
+            mrk_offset = np.fromfile(fid, INT32, 1)[0]
             fid.seek(mrk_offset)
             # skips match_done, meg_to_mri and mri_to_meg
             fid.seek(KIT.INT + (2 * KIT.DOUBLE * 4 ** 2), SEEK_CUR)
-            mrk_count = unpack('i', fid.read(KIT.INT))[0]
+            mrk_count = np.fromfile(fid, INT32, 1)[0]
             pts = []
             for _ in range(mrk_count):
                 # skips mri/meg mrk_type and done, mri_marker

--- a/mne/io/kit/coreg.py
+++ b/mne/io/kit/coreg.py
@@ -15,6 +15,7 @@ from .._digitization import _read_dig_points
 
 
 INT32 = '<i4'
+FLOAT64 = '<f8'
 
 
 def read_mrk(fname):
@@ -44,7 +45,7 @@ def read_mrk(fname):
             for _ in range(mrk_count):
                 # skips mri/meg mrk_type and done, mri_marker
                 fid.seek(KIT.INT * 4 + (KIT.DOUBLE * 3), SEEK_CUR)
-                pts.append(np.fromfile(fid, dtype='d', count=3))
+                pts.append(np.fromfile(fid, dtype=FLOAT64, count=3))
                 mrk_points = np.array(pts)
     elif ext == '.txt':
         mrk_points = _read_dig_points(fname, unit='m')

--- a/mne/io/kit/kit.py
+++ b/mne/io/kit/kit.py
@@ -12,7 +12,6 @@ RawKIT class is adapted from Denis Engemann et al.'s mne_bti2fiff.py.
 from collections import defaultdict, OrderedDict
 from math import sin, cos
 from os import SEEK_CUR, path as op
-from struct import unpack
 
 import numpy as np
 from scipy import linalg
@@ -505,7 +504,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         #
         fid.seek(dirs[KIT.DIR_INDEX_SYSTEM]['offset'])
         # check file format version
-        version, revision = unpack('2i', fid.read(2 * KIT.INT))
+        version, revision = np.fromfile(fid, INT32, 2)
         if version < 2 or (version == 2 and revision < 3):
             version_string = "V%iR%03i" % (version, revision)
             if allow_unknown_format:
@@ -518,31 +517,30 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                     "Set allow_unknown_format=True to load it anyways." %
                     (version_string,))
 
-        sysid = unpack('i', fid.read(KIT.INT))[0]
+        sysid = np.fromfile(fid, INT32, 1)[0]
         # basic info
-        system_name = unpack('128s', fid.read(128))[0].decode()
+        system_name = _read_name(fid, n=128)
         # model name
-        model_name = unpack('128s', fid.read(128))[0].decode()
+        model_name = _read_name(fid, n=128)
         # channels
-        sqd['nchan'] = channel_count = unpack('i', fid.read(KIT.INT))[0]
-        comment = unpack('256s', fid.read(256))[0].decode()
-        create_time, last_modified_time = unpack('2i', fid.read(2 * KIT.INT))
+        sqd['nchan'] = channel_count = int(np.fromfile(fid, INT32, 1)[0])
+        comment = _read_name(fid, n=256)
+        create_time, last_modified_time = np.fromfile(fid, INT32, 2)
         fid.seek(KIT.INT * 3, SEEK_CUR)  # reserved
-        dewar_style = unpack('i', fid.read(KIT.INT))[0]
+        dewar_style = np.fromfile(fid, INT32, 1)[0]
         fid.seek(KIT.INT * 3, SEEK_CUR)  # spare
-        fll_type = unpack('i', fid.read(KIT.INT))[0]
+        fll_type = np.fromfile(fid, INT32, 1)[0]
         fid.seek(KIT.INT * 3, SEEK_CUR)  # spare
-        trigger_type = unpack('i', fid.read(KIT.INT))[0]
+        trigger_type = np.fromfile(fid, INT32, 1)[0]
         fid.seek(KIT.INT * 3, SEEK_CUR)  # spare
-        adboard_type = unpack('i', fid.read(KIT.INT))[0]
+        adboard_type = np.fromfile(fid, INT32, 1)[0]
         fid.seek(KIT.INT * 29, SEEK_CUR)  # reserved
 
         if version < 2 or (version == 2 and revision <= 3):
-            adc_range = float(unpack('i', fid.read(KIT.INT))[0])
+            adc_range = float(np.fromfile(fid, INT32, 1)[0])
         else:
-            adc_range = unpack('d', fid.read(KIT.DOUBLE))[0]
-        adc_polarity, adc_allocated, adc_stored = unpack('3i',
-                                                         fid.read(3 * KIT.INT))
+            adc_range = np.fromfile(fid, FLOAT64, 1)[0]
+        adc_polarity, adc_allocated, adc_stored = np.fromfile(fid, INT32, 3)
         system_name = system_name.replace('\x00', '')
         system_name = system_name.strip().replace('\n', '/')
         model_name = model_name.replace('\x00', '')
@@ -590,7 +588,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         exg_gains = list()
         for i in range(channel_count):
             fid.seek(chan_offset + chan_size * i)
-            channel_type, = unpack('i', fid.read(KIT.INT))
+            channel_type, = np.fromfile(fid, INT32, 1)
             # System 52 mislabeled reference channels as NULL. This was fixed
             # in system 53; not sure about 51...
             if sysid == 52 and i < 160 and channel_type == KIT.CHANNEL_NULL:
@@ -611,7 +609,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                     fid.seek(16, SEEK_CUR)  # misc fields
                     channels[-1]['name'] = _read_name(fid, channel_type)
             elif channel_type in KIT.CHANNELS_MISC:
-                channel_no, = unpack('i', fid.read(KIT.INT))
+                channel_no, = np.fromfile(fid, INT32, 1)
                 fid.seek(4, SEEK_CUR)
                 name = _read_name(fid, channel_type)
                 channels.append({
@@ -647,7 +645,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         # amplifier gain (7)
         #
         fid.seek(dirs[KIT.DIR_INDEX_AMP_FILTER]['offset'])
-        amp_data = unpack('i', fid.read(KIT.INT))[0]
+        amp_data = np.fromfile(fid, INT32, 1)[0]
         if fll_type >= 100:  # Kapper Type
             # gain:             mask           bit
             gain1 = (amp_data & 0x00007000) >> 12
@@ -676,17 +674,17 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         # Acquisition Parameters (8)
         #
         fid.seek(dirs[KIT.DIR_INDEX_ACQ_COND]['offset'])
-        sqd['acq_type'], = acq_type, = unpack('i', fid.read(KIT.INT))
-        sqd['sfreq'], = unpack('d', fid.read(KIT.DOUBLE))
+        sqd['acq_type'], = acq_type, = np.fromfile(fid, INT32, 1)
+        sqd['sfreq'], = np.fromfile(fid, FLOAT64, 1)
         if acq_type == KIT.CONTINUOUS:
-            # samples_count, = unpack('i', fid.read(KIT.INT))
+            # samples_count, = np.fromfile(fid, INT32, 1)
             fid.seek(KIT.INT, SEEK_CUR)
-            sqd['n_samples'], = unpack('i', fid.read(KIT.INT))
+            sqd['n_samples'], = np.fromfile(fid, INT32, 1)
         elif acq_type == KIT.EVOKED or acq_type == KIT.EPOCHS:
-            sqd['frame_length'], = unpack('i', fid.read(KIT.INT))
-            sqd['pretrigger_length'], = unpack('i', fid.read(KIT.INT))
-            sqd['average_count'], = unpack('i', fid.read(KIT.INT))
-            sqd['n_epochs'], = unpack('i', fid.read(KIT.INT))
+            sqd['frame_length'], = np.fromfile(fid, INT32, 1)
+            sqd['pretrigger_length'], = np.fromfile(fid, INT32, 1)
+            sqd['average_count'], = np.fromfile(fid, INT32, 1)
+            sqd['n_epochs'], = np.fromfile(fid, INT32, 1)
             if acq_type == KIT.EVOKED:
                 sqd['n_samples'] = sqd['frame_length']
             else:
@@ -755,7 +753,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                           (sysid,))
         adc_range, adc_stored = LEGACY_AMP_PARAMS[sysid]
     is_meg = np.array([ch['type'] in KIT.CHANNELS_MEG for ch in channels])
-    ad_to_volt = adc_range / (2 ** adc_stored)
+    ad_to_volt = adc_range / (2. ** adc_stored)
     ad_to_tesla = ad_to_volt / amp_gain * channel_gain
     conv_factor = np.where(is_meg, ad_to_tesla, ad_to_volt)
     # XXX this is a bit of a hack. Should probably do this more cleanly at
@@ -765,7 +763,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
     # for the EEG data.
     is_exg = [ch['type'] in (KIT.CHANNEL_EEG, KIT.CHANNEL_ECG)
               for ch in channels]
-    exg_gains /= 2 ** (adc_stored - 14)
+    exg_gains /= 2. ** (adc_stored - 14)
     exg_gains[exg_gains == 0] = ad_to_volt
     conv_factor[is_exg] = exg_gains
     sqd['conv_factor'] = conv_factor[:, np.newaxis]

--- a/mne/io/kit/kit.py
+++ b/mne/io/kit/kit.py
@@ -33,6 +33,11 @@ from ...event import read_events
 from .._digitization import _set_dig_kit
 
 
+FLOAT64 = '<f8'
+UINT32 = '<u4'
+INT32 = '<i4'
+
+
 def _call_digitization(info, mrk, elp, hsp, kit_info):
     # Use values from kit_info only if all others are None
     if mrk is None and elp is None and hsp is None:
@@ -455,10 +460,10 @@ class EpochsKIT(BaseEpochs):
 
 
 def _read_dir(fid):
-    return dict(offset=np.fromfile(fid, np.uint32, 1)[0],
-                size=np.fromfile(fid, np.int32, 1)[0],
-                max_count=np.fromfile(fid, np.int32, 1)[0],
-                count=np.fromfile(fid, np.int32, 1)[0])
+    return dict(offset=np.fromfile(fid, UINT32, 1)[0],
+                size=np.fromfile(fid, INT32, 1)[0],
+                max_count=np.fromfile(fid, INT32, 1)[0],
+                count=np.fromfile(fid, INT32, 1)[0])
 
 
 @verbose
@@ -561,7 +566,8 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         # MGH description: 'acquisition (megacq) VectorView system at NMR-MGH'
         description = \
             f'{system_name} ({sysid}) {full_version} {model_name}'
-        sqd['dtype'] = np.dtype(getattr(np, f'int{adc_allocated}'))
+        assert adc_allocated % 8 == 0
+        sqd['dtype'] = np.dtype(f'<i{adc_allocated // 8}')
 
         # check that we can read this file
         if fll_type not in KIT.FLL_SETTINGS:
@@ -599,7 +605,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                     'type': channel_type,
                     # (x, y, z, theta, phi) for all MEG channels. Some channel
                     # types have additional information which we're not using.
-                    'loc': np.fromfile(fid, dtype='d', count=5),
+                    'loc': np.fromfile(fid, dtype=FLOAT64, count=5),
                 })
                 if channel_type in KIT.CHANNEL_NAME_NCHAR:
                     fid.seek(16, SEEK_CUR)  # misc fields
@@ -616,7 +622,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                 if channel_type in (KIT.CHANNEL_EEG, KIT.CHANNEL_ECG):
                     offset = 6 if channel_type == KIT.CHANNEL_EEG else 8
                     fid.seek(offset, SEEK_CUR)
-                    exg_gains.append(np.fromfile(fid, 'd', 1)[0])
+                    exg_gains.append(np.fromfile(fid, FLOAT64, 1)[0])
             elif channel_type == KIT.CHANNEL_NULL:
                 channels.append({'type': channel_type})
             else:
@@ -632,7 +638,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
         # through unaffected
         fid.seek(dirs[KIT.DIR_INDEX_CALIBRATION]['offset'])
         # (offset [Volt], gain [Tesla/Volt]) for each channel
-        sensitivity = np.fromfile(fid, dtype='d', count=channel_count * 2)
+        sensitivity = np.fromfile(fid, dtype=FLOAT64, count=channel_count * 2)
         sensitivity.shape = (channel_count, 2)
         channel_offset, channel_gain = sensitivity.T
         assert (channel_offset == 0).all()  # otherwise we have a problem
@@ -705,7 +711,7 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
                 # the channel name and its digitized, name, so let's be case
                 # insensitive. It will also prevent collisions with HSP
                 name = name.lower()
-                rr = np.fromfile(fid, 'd', 3)
+                rr = np.fromfile(fid, FLOAT64, 3)
                 if name:
                     assert name not in dig
                     dig[name] = rr
@@ -725,20 +731,20 @@ def get_kit_info(rawfile, allow_unknown_format, standardize_names=None,
             fid.seek(cor_dir['offset'])
             mrk = np.zeros((elp.shape[0] - 3, 3))
             for _ in range(cor_dir['count']):
-                done = np.fromfile(fid, np.int32, 1)[0]
+                done = np.fromfile(fid, INT32, 1)[0]
                 fid.seek(16 * KIT.DOUBLE +  # meg_to_mri
                          16 * KIT.DOUBLE,  # mri_to_meg
                          SEEK_CUR)
-                marker_count = np.fromfile(fid, np.int32, 1)[0]
+                marker_count = np.fromfile(fid, INT32, 1)[0]
                 if not done:
                     continue
                 assert marker_count >= len(mrk)
                 for mi in range(len(mrk)):
                     mri_type, meg_type, mri_done, meg_done = \
-                        np.fromfile(fid, np.int32, 4)
+                        np.fromfile(fid, INT32, 4)
                     assert meg_done
                     fid.seek(3 * KIT.DOUBLE, SEEK_CUR)  # mri_pos
-                    mrk[mi] = np.fromfile(fid, 'd', 3)
+                    mrk[mi] = np.fromfile(fid, FLOAT64, 3)
                 fid.seek(256, SEEK_CUR)  # marker_file (char)
             sqd.update(hsp=hsp, elp=elp, mrk=mrk)
 

